internal/server/server.go

```diff
@@ -70,7 +71,10 @@ func createSOMASServer(
 	}
 
 	availableRules, rulesInPlay := rules.InitialRuleRegistration(gameConfig.IIGOConfig.StartWithRulesInPlay)
-
+	initRoles, err := getNRandValuesFromLstUniqueWherePossible(clientIDs, 3)
+	if err != nil {
+		panic(err)
```

Comment:
Instead of panicking let's change the return type of `createSOMASServer` to `(Server, error)` and return the error.

Error should be `errors.Errrof("Cannot initialise IIGO roles, no clients registered: %v", err)`

Reply:
Hmm... So that means I'll need to return an error from NewSOMASServer, which I would then also need to capture in main_js.go?

Reply:
sorry for asking so much for such a little change. because this is a failing point, it's really important to return good errors. Otherwise you'll just get an error like in the description of #330 

If we make it work with good practices from the start, we'll not need to spend much time debugging it.

I hope you understand :)

Reply:
If you don't want to change the return types I can do that for you. just tell me

Reply:
done in #374 . Do rename the error to `errors.Errrof("Cannot initialise IIGO roles, no clients registered: %v", err)`